### PR TITLE
fix: pin prisma version when generating

### PR DESCRIPTION
**Pull request by ojeytonwilliams**
This is a tad convoluted, but npx defaults to the latest version of prisma and we may not.

Also, `--ignore-scripts` means (finally) Cypress doesn't install the binaries after `pnpm install` which should save a little time.

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

### CODE REVIEW

File: docker/new-api/Dockerfile [hunk 1]
```diff
@@ -17,8 +18,13 @@ COPY --chown=node:node curriculum/ curriculum/
...
 
 RUN pnpm config set dedupe-peer-dependents false
+# While we want to ignore scripts generally, we do need to generate the prisma
+# client. Note: npx is the simplest way to generate the client without us having
+# to have prisma as a prod dependency. The jq tricks are to ensure we're using
+# the right version of prisma.
 RUN pnpm install -F=api -F=curriculum -F tools/scripts/build -F challenge-parser \
-  --frozen-lockfile
+  --frozen-lockfile --ignore-scripts
+RUN cd api && npx prisma@$(jq -r '.devDependencies.prisma' < package.json) generate
```
Comment (naomi-lgbt) [inline docker/new-api/Dockerfile:27]: Out of curiosity why are we using `npx` for this and not `pnpm dlx`?

Comment (ojeytonwilliams) [reply docker/new-api/Dockerfile:27]: To be honest there doesn't seem to be much difference. Ironically, npx is a hair faster, but there's not much in it. Also it's shorter, which I like!

TL;DR: personal preference.

**Review by naomi-lgbt — APPROVED** [review]

**Review by moT01 — APPROVED** [review]
LGTM 🎉
